### ISSUE TITLE
ERROR: Master version must be one of "STABLE" channel supported versions 

### ISSUE THREAD

**Issue by heldersepu** [terraform-google-modules/terraform-google-kubernetes-engine#2177]
### TL;DR

Getting error all of the sudden about an unsupported version

### Expected behavior

there should be no errors and the version should update correctly

### Observed behavior

the error:
```
╷
│ Error: googleapi: Error 400: Master version must be one of "STABLE" channel supported versions [1.28.14-gke.1099000, 1.29.9-gke.1177000, 1.30.5-gke.1014003].
│ Details:
│ [
│   ***
│     "@type": "type.googleapis.com/google.rpc.RequestInfo",
│     "requestId": "0x4d912a749eb45e02"
│   ***
│ ]
│ , badRequest
│ 
│   with module.gke_blockexplorer.google_container_cluster.primary,
│   on .terraform/modules/abc/modules/beta-public-cluster/cluster.tf line 22, in resource "google_container_cluster" "primary":
│   22: resource "google_container_cluster" "primary" ***
│
```

### Terraform Configuration
the relevant part of TF plan shows

```hcl
# module.gke_blockexplorer.google_container_cluster.primary will be updated in-place
  ~ resource "google_container_cluster" "primary" ***
        id                          = "..."
      ~ min_master_version          = "1.30.5-gke.1443001" -> "1.31.1-gke.2105000"
        name                        = "..."
```

### Terraform Version

latest

### Additional information

Not sure where that `1.31.1-gke.2105000` is coming from, we do not have that in our code, there is no `kubernetes_version` set, so the module should be getting the correct one automatically, if I'm following this code correctly:
- https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/modules/beta-public-cluster/cluster.tf#L81
- https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/modules/beta-public-cluster/main.tf#L48-L50
